mesonbuild/dependencies/base.py

```diff
@@ -617,6 +617,8 @@ def search():
         if ret != 0:
             return
 
+        self.is_found = True
+
```

Comment:
I'm not sure to understand how this fix the bug, you're setting it to True earlier now. Are there cases when the try block below actually fails but not with a DependencyException?

Reply:
The try block below will catch any dependency exception and if the dependency is not required, it will set `is_found` to `False` just for the code after that to set it to `True` again unconditionally.

Reply:
ahh, forget me, I mis-read the code. You're right.